src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -107,10 +109,36 @@ public static Email parseEmail(String email) throws ParseException {
     public static Deadline parseDeadline(String deadline) throws ParseException {
         requireNonNull(deadline);
         String trimmedDeadline = deadline.trim();
-        if (!Deadline.isValidDeadline(trimmedDeadline)) {
+        StringBuilder originalInput = new StringBuilder(trimmedDeadline);
+        originalInput.reverse();
+        String[] reversedAndSplit = originalInput.toString().split("\\s+", 2);
+        if (reversedAndSplit.length < 2) {
             throw new ParseException(Deadline.MESSAGE_CONSTRAINTS);
         }
-        return new Deadline(trimmedDeadline);
+        String description = new StringBuilder(reversedAndSplit[1]).reverse().toString();
+        String date = new StringBuilder(reversedAndSplit[0]).reverse().toString();
```

Comment:
Sorry, could you explain why did you choose to reverse the strings before calling split? I find that there's no issue calling split right away :thinking:

Reply:
I was thinking if someone inputs a description including whitespace like, "deadline 1 d/read every single book 1/1/2023", I only want to split the string on the last delimiter. So I reverse the string, split the string on first delimiter then reverse the string again.

Reply:
Ah I see, I didn't think of that, thanks for explaining! :smile: